src/backend/distributed/operations/repair_shards.c

```diff
@@ -98,16 +116,14 @@ master_copy_shard_placement(PG_FUNCTION_ARGS)
 	char *sourceNodeName = text_to_cstring(sourceNodeNameText);
 	char *targetNodeName = text_to_cstring(targetNodeNameText);
 
-	CheckCitusVersion(ERROR);
 	EnsureCoordinator();
+	CheckCitusVersion(ERROR);
 
 	char shardReplicationMode = LookupShardTransferMode(shardReplicationModeOid);
-	if (shardReplicationMode == TRANSFER_MODE_FORCE_LOGICAL)
+	if (doRepair && shardReplicationMode == TRANSFER_MODE_FORCE_LOGICAL)
 	{
 		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("using logical replication in "
-							   "master_copy_shard_placement() requires Citus "
-							   "Enterprise")));
+						errmsg("logical replication cannot be used for repairs")));
```

Comment:
@marcocitus for example here we give this error.

What I am suggesting is erroring out in `LookupShardTransferMode` if it is not `block-writes` mode.

Reply:
but this error is more accurate for this particular case